climakitae/core/data_interface.py

```diff
@@ -1927,70 +1993,75 @@ def _check_valid_input_station(stations, station_options_all):
-                if closest_options is None:
-                    print("Valid options: \n- ", end="")
-                    print("\n- ".join(station_options_all))
-                    raise ValueError("Bad input")
+            # Just one option in the list
+            elif len(closest_options) == 1:
+                print("Closest option: '" + closest_options[0] + "'")
 
-                # Just one option in the list
-                elif len(closest_options) == 1:
-                    print("Closest option: '" + closest_options[0] + "'")
-
-                elif len(closest_options) > 1:
-                    print("Closest options: \n- " + "\n- ".join(closest_options))
+            elif len(closest_options) > 1:
+                print("Closest options: \n- " + "\n- ".join(closest_options))
 
-                print("Outputting data for station='" + closest_options[0] + "'")
-                stations[i] = closest_options[
-                    0
-                ]  # Replace that value in the list with the best option :)
+            print("Outputting data for station='" + closest_options[0] + "'")
+            stations[i] = closest_options[
+                0
+            ]  # Replace that value in the list with the best option :)
 
-                printed_warning = True
+            printed_warning = True
 
         return stations
 
     # Internal functions
-    def _error_handling_warming_level_inputs(wl, argument_name):
-        """Error handling for arguments: warming_level and warming_level_month
+    def _error_handling_warming_level_inputs(
+        wl: Union[list[float], list[int]], argument_name: str
+    ):
+        """
+        Error handling for arguments: warming_level and warming_level_month
         Both require a list of either floats or ints
-        argument_name is either "warming_level" or "warming_level_months" and is used to print an appropriate error message for bad input
+        argument_name is either "warming_level" or "warming_level_months" and is used to
+        print an appropriate error message for bad input
         """
-        if (wl is not None) and (type(wl) != list):
-            if type(wl) in [float, int]:  # Convert float to a singleton list
+        if (wl is not None) and not isinstance(wl, list):
+            if isinstance(wl, (float, int)):  # Convert float to a singleton list
                 wl = [wl]
-            if type(wl) != list:
+            if not isinstance(wl, list):
                 raise ValueError(
-                    "Function argument {0} requires a float/int or list of floats/ints input. Your input: {1}".format(
-                        argument_name, type(wl)
-                    )
+                    f"""Function argument {argument_name} requires a float/int or list 
+                    of floats/ints input. Your input: {type(wl)}"""
                 )
-        if type(wl) == list:
+        if isinstance(wl, list):
```

Comment:
Cool, I just googled why this is a better method for checking the variable type and learned something from your PR. Thanks!

Reply:
Nice! Pro tip: if you install the `pylint` extension in VSCode it'll catch this for you.